### PR TITLE
164 improve testing in takephoto

### PR DESCRIPTION
**Pull request by makfazlic**

### CODE REVIEW

File: app/src/androidTest/java/com/github/mateo762/myapplication/TakePictureTest.kt
Comment (AnirudhhRamesh) [inline app/src/androidTest/java/com/github/mateo762/myapplication/TakePictureTest.kt:1]: Great job commenting your tests
"No habits found" -> Can this be turned into variable/not hardcoded?
Try to use standard line breaks/spacing to make code more readable

File: app/src/main/java/com/github/mateo762/myapplication/profile/ProfileActivity.kt
Comment (AnirudhhRamesh) [inline app/src/main/java/com/github/mateo762/myapplication/profile/ProfileActivity.kt:1]: Great code!

Questions:
Why do you use the if-else on lines 333-337?

Great things:
Nice job once again on commenting your code
Not hardcoding your strings in tests, excellent!

To Improve
line 361, is there a cleaner way to do the "for i in ..."?

File: app/src/androidTest/java/com/github/mateo762/myapplication/TakePictureTest.kt
Comment (makfazlic) [reply app/src/androidTest/java/com/github/mateo762/myapplication/TakePictureTest.kt:1]: When it comes to the hardcoded fields I agree they usually should be turned into a variable. However since it is a test it does not matter. This is what David told me at least.  Standard line breaks/spacing to make code more readable, will do right now.

File: app/src/main/java/com/github/mateo762/myapplication/takephoto/TakePhotoActivity.kt
Comment (AnirudhhRamesh) [inline app/src/main/java/com/github/mateo762/myapplication/takephoto/TakePhotoActivity.kt:1]: l121, great job modularising the code into a function!
Why did you remove the Thread.sleep? Was it perhaps not required in the first place?

Great clean code overall :))

File: app/src/main/res/layout/activity_takephoto.xml
Comment (AnirudhhRamesh) [inline app/src/main/res/layout/activity_takephoto.xml:1]: I believe linear layouts are more efficient, so using the contraint layout to host the two linear layouts is a bit overkill, but otherwise fine :)

**Review by AnirudhhRamesh — APPROVED** [review]
Overview: Overall some great code and test!

Plus:
+ Great job commenting your code
+ Great job modularising your code into functions
+ You make sure not to hardcode strings in your tests, fantastic :))

Negatives:
- for i in 1 to ... (l361 in ProfileActivity) isn't that clean, but ok for this sprint
- ConstraintLayout a bit redundant, you could have gone with LinearLayouts entirely for activity_takephoto.xml

Questions:
- Why do you use if-else on lines 333-337? 

In general great code through, approving!

File: app/src/main/java/com/github/mateo762/myapplication/profile/ProfileActivity.kt
Comment (makfazlic) [reply app/src/main/java/com/github/mateo762/myapplication/profile/ProfileActivity.kt:1]: 1. That was simply a bug from before where we divide by 0. Now I just check if the denominator is 0. Remember the crash in one of the demos.
2. Thanks.
3. Agreed.

File: app/src/main/java/com/github/mateo762/myapplication/takephoto/TakePhotoActivity.kt
Comment (makfazlic) [reply app/src/main/java/com/github/mateo762/myapplication/takephoto/TakePhotoActivity.kt:1]: It was not required there. That is why I removed it. The last one is required due to UX.

File: app/src/main/res/layout/activity_takephoto.xml
Comment (makfazlic) [reply app/src/main/res/layout/activity_takephoto.xml:1]: Agreed. I think it is an overkill, bit it looks nice so I suggest we keep it.
